eng/build.sh

```diff
@@ -339,7 +339,9 @@ if [[ "$restore" == true || "$test_core_clr" == true ]]; then
   install=true
 fi
 InitializeDotNetCli $install
-if [[ "$restore" == true && "$source_build" != true ]]; then
+# Check the dev switch --source-build as well as ensure that source only switches were not passed in via extra properties
+# Source only builds would not have 'dotnet' ambiently available.
+if [[ "$restore" == true && "$source_build" != true && $properties != *"DotNetBuildSourceOnly=true"* ]]; then
```

Comment:
What would actually happen in this scenario though? I get that we already have this condition, but when sourcebuild tries to call restore, won't everything just break since we won't do a restore?

Reply:
The build commands would explicitly use a specific dotnet executable rather than assuming that dotnet is available on the path. But we also don't want to run this anyway even if it was available, because it could introduce prebuilts.

Reply:
It feels a bit brittle to me, but alright, I guess we can cross the bridge of required tools for building when we come to it.